File: BOJ_2309.py
#brute force
#일곱난쟁이

import sys

mini = []
input = sys.stdin.readline


for _ in range (9):
    mini.append(int(input()))
total = sum(mini)

for i in range(9):
    for j in range(i+1,9):
        if 100 == total - (mini[i] + mini[j]): 
            n1,n2=mini[i],mini[j]

            mini.remove(n1)
            mini.remove(n2)
            mini.sort()

            for i in range(len(mini)):
                print(mini[i])
            break

    if len(mini)<9:
        break

File: BOJ_2309.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class BOJ_2309 {
    static List<Integer> child = new ArrayList<>();
    static int[] ans = new int[7];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int sum_init = 0;
        for (int i = 0; i < 9; i++){
            int val =Integer.parseInt(br.readLine());
            sum_init += val;
            child.add(val);
        }

        int total = sum_init;
        aa : for (int i = 0; i < 9; i++){
            for (int j = i+1; j < 9; j++){
                if (100 == total - (child.get(i) + child.get(j))){
                    int n1 = child.get(i);
                    int n2 = child.get(j);

                    child.remove(Integer.valueOf(n1));
                    child.remove(Integer.valueOf(n2));

                    for (int k = 0 ;k < 7; k++){
                        ans[k] = child.get(k);
                    }

                    Arrays.sort(ans);
                    for (int a : ans){
                        System.out.println(a);
                    }
                    break aa;
                }
            }
        }



        br.close();
    }
}
